src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        // generate a specific figure based on the `figureNumber` value
+        switch (figureNumber) {
```

Comment:
lets make 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction